src/python/ensembl/io/genomio/seq_region/rename.py

```diff
@@ -0,0 +1,212 @@
+            )
+    return seq_regions
+
+
+def get_seq_regions_to_replace(
+    session: Session, rename_map: List[SeqRegionReplacement]
+) -> List[SeqRegionReplacement]:
+    """Check each seq_region replacement against the database.
+
+    Args:
+        session: Session from the current core.
+        rename_map: List of remappings to check.
+
+    Returns: same list with the operation changed to the operation it can do.
+
+    """
+
+    seq_regions = []
+    for seqr in rename_map:
+        seqr_stmt = (
+            select(SeqRegion)
+            .select_from(SeqRegion)
+            .join(SeqRegion.seq_region_attrib)
+            .join(SeqRegion.seq_region_synonym)
+            .where(or_(SeqRegion.name == seqr.name, SeqRegionSynonym.synonym == seqr.name))
+        )
+
+        rows = session.execute(seqr_stmt).unique().all()
+        if not rows:
+            logging.warning(f"Seq region not found: {seqr}")
+            seqr.operation = Operation.not_found
+            continue
+        for row in rows:
+            db_seqr: SeqRegion = row[0]
+            if not db_seqr:
+                seqr.operation = Operation.not_found
+            attribs = get_attribs_dict(db_seqr)
+            db_brc_name = attribs.get("BRC4_seq_region_name", "")
+            seqr.old_brc_name = db_brc_name
+            if not db_brc_name:
+                logging.info(f"Seq region {seqr.name} doesn't have a BRC name")
+                seqr.operation = Operation.insert
+                continue
+            if seqr.brc_name == db_brc_name:
+                logging.info(f"Seq region {seqr.name} already exists with same name")
+                seqr.operation = Operation.do_nothing
+                continue
+            logging.info(f"Seq region {seqr.name} already exists as {db_brc_name} instead of {seqr.brc_name}")
+            seqr.operation = Operation.update
+            seqr.seq_region_id = db_seqr.seq_region_id
+            seq_regions.append(seqr)
+
+    return seq_regions
+
+
+def get_attribs(seq_region: SeqRegion) -> List:
+    """Given a seq_region, extract the attribs as value-source items.
+
+    Args:
+        seq_region (SeqRegion): The seq_region from which the attribs are extracted.
+
+    Returns:
+        All attributes as a list of dictionaries with 'value' and 'source' keys.
+    """
+    attribs = seq_region.seq_region_attrib
+    atts = []
+    if attribs:
+        for attrib in attribs:
+            att_obj = {"value": attrib.value, "source": attrib.attrib_type.code}
+            atts.append(att_obj)
+    return atts
+
+
+def get_attribs_dict(seq_region: SeqRegion) -> Dict[str, Any]:
+    """Extracts all the attributes of the given sequence region.
+
+    Args:
+        seq_region: Sequence region.
+
+    Returns:
+        Pairs of source and value for each attribute.
+    """
+
+    attribs = get_attribs(seq_region)
+    attrib_dict = {attrib["source"]: attrib["value"] for attrib in attribs}
+    return attrib_dict
```

Comment:
`get_attribs()` is only called in `get_attribs_dict()`, so I'm wondering if these two can be combined into a single `get_attributes()` and return the dictionary?